### DISCUSSION TITLE
Bot got error broken pipe after replying at certain concurrent

### DISCUSSION THREAD

**Discussion by kingter-sutjiadi** [inspircd/inspircd#2087] [Support]
Hi, 

I have a bot that have a job to reply every client that Privmsg to him, and modifying the message with necessary data for our frontend to process,

but when we tried to run with 100 client that send msg to the bot these happens:

1. we log the bot and he receive the 100 messages from the client perfectly
2. when the bot is trying to reply the messages it only manages to do 60 reply
3. after 60  reply the bot receive this message
" ERROR write tcp 192.168.187.46:51595->13.xxx.xxx.xxx:6667: write: broken pipe : failed to send private message to idn-a771932f-ff1c-47fe-8683-acebb9a20b2a99-1710999968315 with message {"user" ... "
4. when the log shown 60 reply to the client, somehow only 16 client that receive the reply from the bot 

my question is it any configuration that made this thing happen?

here's my config 

```
<connect
        name="noban"
        allow="*"
        parent="main"
        port="6666"
        maxconnwarn="yes"
        useconnectban="no">
<connect
         name="main"
         allow="*"
         hash="&connecthash;"
         password="&connectpassword;"
         maxchans="60"
         timeout="30"
         pingfreq="30"
         hardsendq="3M"
         #hardsendq="1M"
         softsendq="60240"
         #softsendq="10240"
         recvq="60K"
         #recvq="20K"
         threshold="10"
         commandrate="5000"
         #commandrate="1000"
         fakelag="no"
         localmax="200"
         globalmax="200"
         resolvehostnames="no"
         useident="no"
         usednsbl="&usednsbl;"
         limit="20000">
```

Thank you

**Comment by sadiepowell** [comment]
Its probably hitting some kind of limit. If you add a log tag something like `<log type="ANNOUNCEMENT REMOTEANNOUNCEMENT USERS SOCKET" level="debug" target="users.log" flush="1" method="file">` it'll log why the bot is being disconnected.

**Reply by kingter-sutjiadi** [reply]
hi @SadieCat , thank you for response,

may i ask where do i put that log tag is it on config?

**Reply by sadiepowell** [reply]
It goes in your config yes. Make sure to rehash or restart after adding it.

**Reply by kingter-sutjiadi** [reply]
thank you @SadieCat will try it

**Reply by kingter-sutjiadi** [reply]
i'm receiving this error  after the log enabled

```
Mon Mar 25 2024 11:41:35 SOCKET: BufferedSocket::DoConnect success
Mon Mar 25 2024 11:41:35 SOCKET: Error on FD 7 - 'received ERROR Server irc-2.xxxx.com already exists on server irc-1.xxxx.com!'
Mon Mar 25 2024 11:41:35 SOCKET: Remove file descriptor: 7
Mon Mar 25 2024 11:42:05 SOCKET: New file descriptor: 7
Mon Mar 25 2024 11:42:05 SOCKET: BufferedSocket::DoConnect success
Mon Mar 25 2024 11:42:05 SOCKET: Error on FD 7 - 'received ERROR Server irc-2.xxxx.com already exists on server irc-1.xxxx.com!'
Mon Mar 25 2024 11:42:05 SOCKET: Remove file descriptor: 7
Mon Mar 25 2024 11:42:35 SOCKET: New file descriptor: 7
Mon Mar 25 2024 11:42:35 SOCKET: BufferedSocket::DoConnect success
Mon Mar 25 2024 11:42:35 SOCKET: Error on FD 7 - 'received ERROR Server irc-2.xxxx.com already exists on server irc-1.xxxx.com!'
Mon Mar 25 2024 11:42:35 SOCKET: Remove file descriptor: 7
Mon Mar 25 2024 11:43:05 SOCKET: New file descriptor: 7
Mon Mar 25 2024 11:43:05 SOCKET: BufferedSocket::DoConnect success
Mon Mar 25 2024 11:43:05 SOCKET: Error on FD 7 - 'received ERROR Server irc-2.xxxx.com already exists on server irc-1.xxxx.com!'
Mon Mar 25 2024 11:43:05 SOCKET: Remove file descriptor: 7
```

anyway i'm running multiple instance  and using haproxy in front of the server

**Reply by kingter-sutjiadi** [reply]
also from my BOT client receive this error message:

``` Received : ERROR :Closing link: (IDNSystem@3.1.239.178) [RecvQ exceeded]   [36mpackage[0m=server.irc```

i've increased the number of recvq at the config but still got the error on top

**Reply by sadiepowell** [reply]
I don't know what you're using it for exactly but if you're sending a lot of data you're probably going to hit the limit eventually. If you're really sure that you want to unlimit your bot you might want to make it an oper with the `users/flood/increased-buffers` privilege instead.

https://docs.inspircd.org/3/configuration/#class

**Reply by kingter-sutjiadi** [reply]
thank you @SadieCat will try it
